### PR TITLE
[HOTFIX] Fix ArrayOutOfBound exception when duplicate measure in projection column

### PR DESCRIPTION
**Pull request by ajantha-bhat**
problem: ArrayOutOfBound exception when duplicate measure in the projection column

cause: In query executor, when the reusable buffer is formed. It was considering only the unique values. Need to consider all the projections.

solution: consider all the projections, while forming a reusable buffer.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed? NA
 
 - [ ] Any backward compatibility impacted? NA
 
 - [ ] Document update required? NA

 - [ ] Testing done
        yes, updated UT.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.  NA

### CODE REVIEW

**Comment by ajantha-bhat** [comment]
@kunal642 , @kumarvishal09 : please check this

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/1582/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/1795/

File: store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java [hunk 1]
```diff
@@ -575,7 +575,7 @@ public void testReadColumnTwice() throws IOException, InterruptedException {
 
     CarbonReader reader = CarbonReader
         .builder(path, "_temp")
-        .projection(new String[]{"name", "name", "age", "name"})
+        .projection(new String[]{"name", "age", "age", "name"})
```
[the remark this answers is not in the document]
Comment (ajantha-bhat) [reply store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java]: @qiuchenjian : NO, age is an int column (measure), check schema in writer buildler. 

No need of new test case, now this test case handle testing duplicate measures and dimensions.

[the remark this answers is not in the document]
Comment (qiuchenjian) [reply store/sdk/src/test/java/org/apache/carbondata/sdk/file/CarbonReaderTest.java]: OK, I just noticed this  “fields[1] = new Field("age", DataTypes.INT);”

**Comment by ravipesala** [comment]
LGTM

**Comment by kunal642** [comment]
LGTM
